csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -142,14 +143,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                     {
                         Debug.Assert(response.Protocol == Protocol.Ice2);
 
-                        // Read the reply status byte immediately after the encapsulation header; + 2 corresponds to the
-                        // encoding in the header.
-                        byte b = response.Payload[1 + sizeLength + 2];
-                        ReplyStatus replyStatus = b.AsReplyStatus();
+                        // Read the reply status byte immediately after the encapsulation header; +1 correspond to the
+                        // result type and +2 corresponds to the encoding in the encapsulation header.
+                        ReplyStatus replyStatus = response.Payload[1 + sizeLength + 2].AsReplyStatus();
 
-                        buffer[0] = b;
+                        buffer[0] = (byte)replyStatus;
                         if (replyStatus == ReplyStatus.UserException)
                         {
+                            // We are forwarding an user exception encoded using 1.1 encoding and ice2 protocol over
```

Comment:
Suggestion: We are forwarding a user exception encoded using 1.1 received over ice2 to ice1. Note that we never perform any transcoding.